tests/queries/0_stateless/02377_quota_key.sh

```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+# Tags: no-parallel
+
+CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+# shellcheck source=../shell_config.sh
+. "$CURDIR"/../shell_config.sh
+
+${CLICKHOUSE_CLIENT} -q "DROP USER IF EXISTS u_02377"
+${CLICKHOUSE_CLIENT} -q "drop quota if exists q_02377"
+${CLICKHOUSE_CLIENT} -q "CREATE USER u_02377 IDENTIFIED WITH plaintext_password BY 'password';"
+${CLICKHOUSE_CLIENT} -q "CREATE QUOTA q_02377 KEYED BY client_key FOR INTERVAL 1 month MAX queries = 100 TO u_02377;"
+
+${CLICKHOUSE_CLIENT} --user=u_02377 --password=password --quota_key=q_02377 --query="select 1"
```

Comment:
How does it check that the quota_key is successfully applied?

Maybe make queries in a loop to get error, then change quota key and check that you can make at least one more query successfully?

Reply:
And check the existing tests for quota_key via HTTP.

Reply:
This quota requires a client supplied key (`KEYED BY client_key`) - this is what we are checking here - if key is not supplied, then connect will produce an exception (`QUOTA_REQUIRES_CLIENT_KEY`) and test will fail.

Reply:
```
yakov@ClickHouse-ubuntu:~/ClickHouse-my-quota/build_debug/programs$ ./clickhouse client -u editor_api --password password
ClickHouse client version 22.8.1.1.
Connecting to localhost:9000 as user editor_api.
Connected to ClickHouse server version 22.8.1 revision 54458.

Cannot load data for command line suggestions: Code: 496. DB::Exception: Received from localhost:9000. DB::Exception: Quota editor_api (for user editor_api) requires a client supplied key.. (QUOTA_REQUIRES_CLIENT_KEY) (version 22.8.1.1)
ClickHouse-ubuntu :) SELECT 1

SELECT 1

Query id: 68fd26e7-48b3-4c1f-b740-4bb8726b5177

Connecting to localhost:9000 as user editor_api.
Connected to ClickHouse server version 22.8.1 revision 54458.


0 rows in set. Elapsed: 0.002 sec.

Received exception from server (version 22.8.1):
Code: 496. DB::Exception: Received from localhost:9000. DB::Exception: Quota editor_api (for user editor_api) requires a client supplied key.. (QUOTA_REQUIRES_CLIENT_KEY)
```

Reply:
Ok.